indico/queries/submission.py

```diff
@@ -124,6 +125,58 @@ def process_response(self, response) -> Submission:
         return Submission(**(super().process_response(response)["submission"]))
 
 
+class WaitForSubmissions(RequestChain):
+    """
+    Given submission_ids, wait for all to finish processing
+    """
+
+    query = """
+        query ListSubmissions(
+            $submissionIds: [Int],
+        ){
+            submissions(
+                submissionIds: $submissionIds,
+            ){
+                submissions {
+                    id
+                    datasetId
+                    workflowId
+                    status
+                    inputFile
+                    inputFilename
+                    resultFile
+                    retrieved
+                    errors
+                }
+            }
+        }
+    """
+
+    def __init__(
+        self, submission_ids: List[int], timeout: Union[int, float] = 60,
+    ):
+        if not submission_ids:
+            raise IndicoInputError("Please provide submission ids")
+
+        self.submission_ids = submission_ids
+        self.timeout = timeout
+        self.status_check = partial(ne, "PROCESSING")
+        self.status_getter = partial(ListSubmissions, self.submission_ids, limit=None)
+
```

Comment:
We wouldn't happen to have query for just status would we?

Reply:
nope - just GetSubmission for the whole object and ListSubmissions for multiple

Reply:
We could make a hidden query here to reduced the fields requested (since list gets a lot more). This would only get you up to Fog where Submission status resolver would have to be written with this in mind. We could make the first portion of making the client ask for less and the second portion if we detect an issue server side with these queries. 

Additionally I think eventually we need to start having get multiple queries that don't use the list method. Andy asked for this as he pointed out they get really annoying with the paging if you aren't doing paging and just getting items by ids. But that's a tangent I think for this one we just start with a GetSubmissionStatuses query that can take a list of ids and return objects with just the status, id, and errors filed in.

Reply:
hmmm, for Andy... you don't have to paginate - could just not provide a `limit`
maybe it's part of a bigger ask to filter without limits? not sure... but ListSubmissions does work without pagination

As for here - usually when we have these "limited field" queries, we need a server-side query to support it first (since the graphql query name is different) before we use it in the client. I can make the server-side query if we'd like - my thinking was that a submission "row" is small enough as is that querying for the whole thing isn't too bad, but can see how that might cause huge issues later with the high load we're seeing.

Reply:
It's more that the nested object is a pain to deal with particularly how it affects their caching layer. He can explain if you ask him.